internal/pkg/skuba/addons/cilium_test.go

```diff
@@ -0,0 +1,286 @@
+				return
+			}
+		})
+	}
+}
+
+func Test_renderContext_CiliumOperatorImage(t *testing.T) {
+	type test struct {
+		name          string
+		renderContext renderContext
+		want          string
+	}
+	for _, ver := range kubernetes.AvailableVersions() {
+		tt := test{
+			name: "render Cilium Operator Image URL when cluster version is " + ver.String(),
+			renderContext: renderContext{
+				addon: Addon{
+					addon:             kubernetes.Cilium,
+					templater:         renderCiliumTemplate,
+					callbacks:         nil,
+					addonPriority:     normalPriority,
+					getImageCallbacks: []getImageCallback{GetCiliumImage},
+				},
+				config: AddonConfiguration{
+					ClusterVersion: ver,
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+			},
+			want: img.ImageRepository + "/cilium-operator:([[:digit:]]{1,}.){2}[[:digit:]]{1,}(-rev[:digit:]{1,})?",
+		}
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.renderContext.CiliumOperatorImage()
+			matched, err := regexp.Match(tt.want, []byte(got))
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			if !matched {
+				t.Errorf("renderContext.CiliumOperatorImage() = %v, want %v", got, tt.want)
+				return
+			}
+		})
+	}
+}
+
+func Test_renderContext_CiliumImage(t *testing.T) {
+	type test struct {
+		name          string
+		renderContext renderContext
+		want          string
+	}
+	for _, ver := range kubernetes.AvailableVersions() {
+		tt := test{
+			name: "render Cilium Image URL when cluster version is " + ver.String(),
+			renderContext: renderContext{
+				addon: Addon{
+					addon:             kubernetes.Cilium,
+					templater:         renderCiliumTemplate,
+					callbacks:         nil,
+					addonPriority:     normalPriority,
+					getImageCallbacks: []getImageCallback{GetCiliumImage},
+				},
+				config: AddonConfiguration{
+					ClusterVersion: ver,
+					ControlPlane:   "",
+					ClusterName:    "",
+				},
+			},
+			want: img.ImageRepository + "/cilium:([[:digit:]]{1,}.){2}[[:digit:]]{1,}(-rev[:digit:]{1,})?",
+		}
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.renderContext.CiliumImage()
+			matched, err := regexp.Match(tt.want, []byte(got))
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			if !matched {
+				t.Errorf("renderContext.CiliumImage() = %v, want %v", got, tt.want)
+				return
+			}
+		})
+	}
+}
+
+func Test_ciliumCallbacks_beforeApply(t *testing.T) {
+	type test struct {
+		name               string
+		addonConfiguration AddonConfiguration
+		skubaConfiguration *skuba.SkubaConfiguration
+		wantErr            bool
+	}
+	for _, ver := range kubernetes.AvailableVersions() {
+		tt := test{
+			name: "Test Cilium callbacks beforeApply when cluster version is " + ver.String(),
+			addonConfiguration: AddonConfiguration{
+				ClusterVersion: ver,
+				ControlPlane:   "",
+				ClusterName:    "",
+			},
+			skubaConfiguration: &skuba.SkubaConfiguration{
+				AddonsVersion: kubernetes.AddonsVersion{},
+			},
```

Comment:
`beforeApply()` does not need this, maybe we could drop this block.

Reply:
removed